test/testCSVParser.cc

```diff
@@ -128,7 +128,45 @@ SUITE(CSVParser)
       reportFromCSVFile(isn,osn,*this);
       
       CHECK(osn.str().find("missing numerical data") != std::string::npos);
-    }    
+    }
+    
+   TEST_FIXTURE(CSVDialog,guessSpaceFile)
+    {
+      spec=DataSpec();
+      string url="../test/testEqGodley.csv";
+      // Parse path to input file. 
+      boost::regex ex(R"(^(.+)\/([^/]+)$)");      
+      boost::cmatch what;
+      if (regex_match(url.c_str(), what, ex)) {
+       // what[0] contains the whole string 	 
+       // what[1] is the directory path
+       // what[2] is the fragment   
+      } else throw runtime_error("Failure to match file: "+url);            
```

Comment:
I don't understand what you're doing here.

Reply:
If you need to test loading a file url, you can use the getcwd() system call to obtain the full path to the current working directory, then construct a URL like string("file://")+getcwd(buffer)+"/testEqGodley.csv".

Reply:
> I don't understand what you're doing here.

Because I didn't know the unit tests are executed from a random directory, I parsed the route to the CSV file. But it is no longer necessary :)


> If you need to test loading a file url, you can use the getcwd() system call to obtain the full path to the current working directory, then construct a URL like string("file://")+getcwd(buffer)+"/testEqGodley.csv".

Thank you, that is a much simpler solution to what I had. However, because of what you explained in a previous post, the parsing of the file path is no longer necessary. I'm just running the coverage test again and will commit the code with the suggested changes as soon as it is done.